include/aspect/parameters.h

```diff
@@ -97,6 +97,38 @@ namespace aspect
              false;
     }
 
+    /**
+     * A struct that contains the enums to decide what to do when a nonlinear solver fails.
+     */
+    struct NonlinearSolverFailureStrategy
```

Comment:
This might be more easily done with a namespace. Your `struct` does not have data members.

Reply:
I recognize that there are other structs of the same kind. I'm going to fix those too right next.

Reply:
Ah, drats. These are class members. You can't have a namespace be a class member :-( (That's one of my long-running complaints about C++...)